### PR TITLE
feat(libs): move profile client to libs

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Only have one implementation of a profile server client, instead of various implementations in the services where its required.

## This pull request

- Moves the profile client implementation from auth-server and into a dedicated library in libs

## Issue that this pull request solves

Closes: #FXA-9274

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/bin/key_server.js [hunk 1]
```diff
@@ -31,19 +31,14 @@ const { Container } = require('typedi');
...
-} = require('../lib/types');
+const { AuthLogger, AuthFirestore, AppConfig } = require('../lib/types');
 const { setupFirestore } = require('../lib/firestore-db');
 const { AppleIAP } = require('../lib/payments/iap/apple-app-store/apple-iap');
 const { AccountEventsManager } = require('../lib/account-events');
 const { AccountDeleteManager } = require('../lib/account-delete');
 const { gleanMetrics } = require('../lib/metrics/glean');
 const Customs = require('../lib/customs');
-const Profile = require('../lib/profile/client');
+const { ProfileClient } = require('@fxa/profile/client');
```
Comment (julianpoy) [inline packages/fxa-auth-server/bin/key_server.js:41]: I forget - this type of import will work fine in all envs now, right?

File: packages/fxa-auth-server/bin/key_server.js [hunk 2]
```diff
@@ -211,11 +206,16 @@ async function run(config) {
...
-  const profile = new Profile(log, config, error, statsd);
+  const profile = new ProfileClient(log, {
+    ...config.profileServer,
+    serviceName: 'subhub',
+  });
   Container.set(ProfileClient, profile);
   const bounces = require('../lib/bounces')(config, database);
   const senders = await require('../lib/senders')(log, config, bounces, statsd);
 
+  //await profile.updateDisplayName('6b963c6d287441b6b17686b339d28a17', 'Peter peter');
```
Comment (julianpoy) [inline packages/fxa-auth-server/bin/key_server.js:217]: Remove?

Comment (StaberindeZA) [reply packages/fxa-auth-server/bin/key_server.js:217]: Lol yup! Thanks

File: packages/fxa-auth-server/bin/key_server.js (hunk 1 above)
Comment (StaberindeZA) [reply packages/fxa-auth-server/bin/key_server.js:41]: Yup. In the case of `fxa-auth-server` as long as the file is picked up by the build process, which runs `tsc-alias`, it should be good to go.

**Review by julianpoy — APPROVED** [review]

**Pull request merged**